### PR TITLE
PRO-1219 PRO-1257 Contributors can "discard draft" properly via the on-page kebab menu

### PR DESCRIPTION
**Pull request by boutell**
Fetch draft properly when refreshing context bar after discarding draft; remove beforeMove handlers made obsolete and problematic by the decision to just autopublish moves (those scenarios no longer apply and the handlers caused the original bug of this ticket); remove redundant logic for discard draft messaging; return doc correctly from discard draft; consistent messaging

### CODE REVIEW

**Comment by boutell** [comment]
(Nothing here is changelog-worthy because contributors are a new beast.)

File: modules/@apostrophecms/page-type/index.js [hunk 1]
```diff
@@ -154,23 +154,6 @@ module.exports = {
           }
         }
       },
-      afterPublish: {
```
Comment (abea) [inline modules/@apostrophecms/page-type/index.js:157]: Is this related to the in-context discard draft change? If not, it's a confusing inclusion.

Comment (boutell) [reply modules/@apostrophecms/page-type/index.js:157]: I believe I needed this fix to get the other to work.

This code was removed because moving a page now automatically reflects in the other mode, so this "replay it on publish" code is redundant, and also harmful.

**Review by abea — APPROVED** [review]
